### PR TITLE
[WIP] Alerting system design doc

### PR DESCRIPTION
**Pull request by nathanielc**
We should probably flesh out this doc with more of the existing alert design. For now this doc contains the proposed enhancements to the alerting system.
 
@joelegasse

### CODE REVIEW

**Comment by nathanielc** [comment]
A section on the write path has been added.

**Comment by nathanielc** [comment]
@joelegasse I have updated this doc with information about existing alerts and this new alert system and some of the differences.

File: ALERT_DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        {"pagerDuty": {"serviceKey":"XXX"}}
+    ]
+}
+```
+
+
+#### Implementation
+
+The underlying implementation will be a basic publish/subscribe system.
+Various subscriber can be defined via the above definitions which can in turn publish back to the internal system or send alerts to third parties.
```
Comment (nathanielc) [inline ALERT_DESIGN.md]: The actual definitions need to be worked out and documented. The above are just example ideas.
Somethings not addressed by the above are how alerts are referenced, via a name/ID or something else.

**Comment by dstreppa** [comment]
+1

**Pull request merged**